### PR TITLE
Add debugging construcstors to LeakTrackingTestConfig, per leak type.

### PR DESCRIPTION
**Pull request by polina-c**

### CODE REVIEW

File: pkgs/leak_tracker_flutter_testing/CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,7 @@
+# 1.0.2
+
+* Add debugging construcstors to LeakTrackingTestConfig, per leak type.
```
Comment (justinmc) [inline pkgs/leak_tracker_flutter_testing/CHANGELOG.md:3]: "construcstors" => "constructors"

File: pkgs/leak_tracker_flutter_testing/lib/src/model.dart [hunk 2]
```diff
@@ -57,7 +57,36 @@ class LeakTrackingTestConfig {
...
+    this.notGCedAllowList = const <String, int>{},
+    this.notDisposedAllowList = const <String, int>{},
+    this.allowAllNotDisposed = false,
+    this.allowAllNotGCed = false,
+  });
+
+  /// Creates a new instance for debugging notDisposed leaks.
+  ///
+  /// This configuration will collect stack traces on start and disposal,
+  /// and retaining path for notGCed objects.
```
Comment (justinmc) [inline pkgs/leak_tracker_flutter_testing/lib/src/model.dart:78]: "retaining" => "retain"

**Review by justinmc — APPROVED** [review]
LGTM :+1:  I'm no expert on the leak tracker but looks fine at a high level!

File: pkgs/leak_tracker_flutter_testing/lib/src/model.dart (hunk 2 above)
Comment (polina-c) [reply pkgs/leak_tracker_flutter_testing/lib/src/model.dart:78]: `retaining path` is a concept in memory management: https://docs.flutter.dev/tools/devtools/memory#root-object-retaining-path-and-reachability

replaced with 'the retaining path' to show it is noun.

Comment (justinmc) [reply pkgs/leak_tracker_flutter_testing/lib/src/model.dart:78]: Ah got it, thanks!

**Review by CoderDake — APPROVED** [review]

**Pull request merged**